File: Loginapp_python/Clases/Usuarios.py
class Usuario:
    def __init__(self, nombre, apellido, email, username, clave, rol):
        self.nombre = nombre
        self.apellido = apellido
        self.email = email
        self.username = username
        self.clave = clave
        self.rol = rol

from Clases.Conector import Conector

class Usuarios:

    def __init__(self):
        self.db = Conector()
    
    def validar_usuarios(self, username, clave):
        sql = "SELECT * from usuarios where username = %s AND clave = %s"
        values = (username, clave)
        return self.db.select(sql, values)
    
    def validarUsuario(self, username, clave):
        return self.validar_usuarios(username, clave)
    
    def registrarUsuarios(self, nombre, apellido, email, username, clave, rol):
        sql = "INSERT INTO usuarios (nombre, apellido, email, username, clave, rol) VALUES (%s, %s, %s, %s, %s, %s)"
        values = (nombre, apellido, email, username, clave, rol)
        return self.db.execute_query(sql, values)

File: AplicacionMysql/AplicacionMysql/src/Clases/Usuarios.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Clases;

import java.sql.PreparedStatement;
import java.sql.SQLException;

/**
 *
 * @author Luna Caes
 */
public class Usuarios {
    
    private final Conector con;
    
    public Usuarios(){
        con = new Conector();
    }
    
    public void insertarUsuarios(String nombre, String apellido, String email, String username, String clave, String rol) throws SQLException{
        
        String sql = "INSERT INTP usuarios (nombre, apellido, email, username, clave, rol) VALUES (?,?,?,?,?,?)";
        
       try(PreparedStatement ps = con.prepararStatement(sql)){
        ps.setString(1, nombre);
        ps.setString(2, apellido);
        ps.setString(3, email);
        ps.setString(4, username);
        ps.setString(5, clave);
        ps.setString(6, rol);
        ps.execute();
        
        System.out.println("Usuarios registrado exitosamente");
    } catch(SQLException e){
        System.out.println("Error al insertar usuario");
    } finally{
           con.desconectar();
       }
}
}
